serialapi/factory_reset.go

```diff
@@ -0,0 +1,24 @@
+package serialapi
+
+import (
+	"time"
+
+	"github.com/gozwave/gozw/protocol"
+	"github.com/gozwave/gozw/session"
+)
+
+// FactoryReset will clear the network configuration and Soft Reset the controller
+// WARNING: This can (and often will) cause the device to get a new USB address,
+// rendering the serial port's file descriptor invalid.
+func (s *Layer) FactoryReset() {
+
+	request := &session.Request{
+		FunctionID: protocol.FnSetDefault,
+		HasReturn:  false,
+	}
+
+	s.sessionLayer.MakeRequest(request)
+
+	time.Sleep(1500 * time.Millisecond)
```

Comment:
This sounds familiar from the spec, is that where this comes from?

Reply:
Yes! I actually looked at how another go zwave wrapper did it and they just reset the network configuration and then the device. Looking into the spec `SetDefault` is the function that resets all configuration.